src/lib/data/data_loader.ts

```diff
@@ -45,4 +51,45 @@ export class DataModel {
   async addBudget(budget: Budget): Promise<void> {
     return this.database.addDocument(Collections.Budgets, budget);
   }
+
+  //These should be moved
+  async addUser(email: string, password: string, user: User): Promise<void> {
+    const auth = getAuth();
+    createUserWithEmailAndPassword(auth, email, password)
```

Comment:
The user needs to get added in two places (which is what you have) but with the same id.  So, `createUserWithEmailAndPassword` will create a new user in a special Firebase user database, and it will assign a random ID to that user.  When you log in, you can get the ID of the currently logged in user via `auth.currentUser`.

But then since there's other data about the user we need to store, we also need to add the user to the Firestore database, which is what you do in the line `this.database.addDocument(Collections.Users, user);`  does. However, that document needs to use the same ID that was created with `createUserWithEmailAndPassword`.  Currently, it just uses whatever was in the `id` field of the `user: User` parameter.  So before adding the user to the Firestore database (`return this.database.addDocument(...)`), you need to set `user.id = auth.currentUser`.  And that should work because `createUserWithEmailAndPassword` automatically logs in the new user.

Reply:
will do after the demo